src/Estimators/include/BipedalLocomotion/RobotDynamicsEstimator/SubModelKinDynWrapper.h

```diff
@@ -0,0 +1,165 @@
+ * @copyright 2023 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the BSD-3-Clause license.
+ */
+
+#ifndef BIPEDAL_LOCOMOTION_ESTIMATORS_SUBMODELKINDYNWRAPPER_H
+#define BIPEDAL_LOCOMOTION_ESTIMATORS_SUBMODELKINDYNWRAPPER_H
+
+#include <map>
+
+// iDynTree
+#include <iDynTree/Core/EigenHelpers.h>
+#include <iDynTree/KinDynComputations.h>
+#include <iDynTree/Model/FreeFloatingState.h>
+#include <iDynTree/Model/Model.h>
+
+// Eigen
+#include <Eigen/Dense>
+
+// RDE
+#include <BipedalLocomotion/RobotDynamicsEstimator/SubModel.h>
+
+namespace BipedalLocomotion
+{
+namespace Estimators
+{
+namespace RobotDynamicsEstimator
+{
+
+/**
+ * SubModelKinDynWrapper is a concrete class and implements a wrapper of the KinDynComputation class
+ * from iDynTree. The class is used to take updated the sub-model kinematics and dynamics
+ */
+
+class SubModelKinDynWrapper
+{
+
+private:
+    SubModel subModel; /**< SubModel struct containing all the information about the sub-model */
+    iDynTree::KinDynComputations kinDyn; /**< kindyncomputation object used to get the sub-model
+                                            kinematic and dynamic information */
+    iDynTree::MatrixDynSize massMatrix; /**< Mass matrix of the sub-model */
+    iDynTree::FreeFloatingGeneralizedTorques genForces; /**< Generalized force vector */
+    std::map<std::string, iDynTree::MatrixDynSize> jacFTList; /**< Jacobians of the FT sensors */
+    std::map<std::string, iDynTree::MatrixDynSize> jacAccList; /**< Jacobians of the accelerometer
+                                                                  sensors */
+    std::map<std::string, iDynTree::MatrixDynSize> jacGyroList; /**< Jacobians of the gyroscope
+                                                                   sensors */
+    std::map<std::string, iDynTree::MatrixDynSize> jacContactList; /**< Jacobians of the external
+                                                                      contacts */
+    std::map<std::string, iDynTree::Vector6> dJnuList; /**< Accelerometer bias accelerations */
+    std::map<std::string, iDynTree::Matrix3x3> accRworldList; /**< Rotation matrix of the
+                                                                 accelerometer frame wrt world */
+
+protected:
+    /**
+     * updateDynamicsVariableState updates the value of all the member variables containing
+     * information about the robot kinematics and dynamics
+     */
+    bool updateDynamicsVariableState(std::shared_ptr<iDynTree::KinDynComputations> kinDynFullModel);
+
+public:
+    /**
+     * @brief initialize initializes the kindyncomputation object and resizes
+     * all the variable not allocated yet.
+     * @param subModel a struct of type SubModif el describing the subModel structure and model.
+     * @return a boolean value saying if all the variables are initialized correctly.
+     */
+    bool initialize(SubModel subModel);
```

Comment:
You can pass the ``subModel`` here as const reference to avoid an intermediary copy